src/averageAges.js

```diff
@@ -55,7 +71,22 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  let children = people
+    .filter(kid => people.some(mother => kid.mother === mother.name));
+
+  const sons = children.filter(son => son.sex === 'm');
+
+  children = !onlyWithSon ? children : sons;
```

Comment:
suggestion: here, I'd rather combine filtering conditions so that you decide right away where that's a son or not, by that avoid using ternary, that definitely looks redundant